src/history.js

```diff
@@ -40,7 +40,7 @@ module.exports = function createHistory (options, onChange) {
   }
 
   function url () {
-    if (mode === 'history') return location.pathname + location.search
+    if (mode === 'history') return location.pathname + location.search + '#' + getHash()
```

Comment:
Can you make it so that # is only added if hash is not empty?

Reply:
👍 good catch